lpc82x-hal/src/lib.rs

```diff
@@ -472,7 +448,7 @@ impl Peripherals {
         Peripherals {
             // HAL peripherals
             DMA   : DMA::new(p.DMA),
-            GPIO  : GPIO::new(p.GPIO_PORT),
+            GPIO  : unsafe { GPIO::new(p.GPIO_PORT) },
```

Comment:
Would be nice to add a comment (maybe above `Peripherals {`) to explain that any usage of unsafe constructors here is safe, as the hardware is in its initial state, which is what the constructor assumes.